app/code/community/Hps/Securesubmit/Model/Payment.php

```diff
@@ -252,6 +243,29 @@ private function _authorize(Varien_Object $payment, $amount, $capture)
         return $this;
     }
 
+    /**
+     * @param Varien_Object|Mage_Sales_Model_Order_Payment  $payment
+     * @param float                                         $amount
+     * @param HpsReportTransactionDetails|HpsAuthorization  $response
+     * @param Mage_Payment_Model_Method_Abstract::STATUS_UNKNOWN|STATUS_APPROVED|STATUS_ERROR|STATUS_DECLINED|STATUS_VOID|STATUS_SUCCESS                                   $status
+     */
+    protected function closeTransaction($payment, $amount, $response, $status = self::STATUS_APPROVED){
+        $payment->setStatus($status);
+        $payment->setAmount($amount);
+        if (property_exists($response,'cardType')){
+            $payment->setCcType($response->cardType);
```

Comment:
are you suggesting portico does not identify the card type correctly?
That second question i honestly didn't investigate. it sounds as if your hinting that it might. I will check